openquake/hazardlib/gsim/parker_2020.py

```diff
@@ -0,0 +1,752 @@
+    """
+    For the CAM North region.
+    """
+    def __init__(self):
+        super(ParkerEtAl2020SSlabCAMN, self) \
+            .__init__(region="CAM", saturation_region="CAM_N")
+
+
+class ParkerEtAl2020SSlabCAMS(ParkerEtAl2020SSlab):
+    """
+    For the CAM South region.
+    """
+    def __init__(self):
+        super(ParkerEtAl2020SSlabCAMS, self) \
+            .__init__(region="CAM", saturation_region="CAM_S")
+
+
+class ParkerEtAl2020SSlabCascadia(ParkerEtAl2020SSlab):
+    """
+    Cascadia, other mapped basin (Tacoma, Everett, Georgia, etc.).
+    """
+
+    REQUIRES_SITES_PARAMETERS = {'vs30', 'z2pt5'}
+
+    def __init__(self):
+        super(ParkerEtAl2020SSlabCascadia, self).__init__(region="Cascadia")
+
+
+class ParkerEtAl2020SSlabCascadiaOut(ParkerEtAl2020SSlab):
+    """
+    Cascadia, estimate of Z2.5 outside mapped basin.
+    """
+
+    REQUIRES_SITES_PARAMETERS = {'vs30', 'z2pt5'}
+
+    def __init__(self):
+        super(ParkerEtAl2020SSlabCascadiaOut, self) \
+            .__init__(region="Cascadia", basin="out")
+
+
+class ParkerEtAl2020SSlabCascadiaSeattle(ParkerEtAl2020SSlab):
+    """
+    Cascadia, Seattle basin.
+    """
+
+    REQUIRES_SITES_PARAMETERS = {'vs30', 'z2pt5'}
+
+    def __init__(self):
+        super(ParkerEtAl2020SSlabCascadiaSeattle, self) \
+            .__init__(region="Cascadia", basin="Seattle")
+
+
+class ParkerEtAl2020SSlabJapanPac(ParkerEtAl2020SSlab):
+    """
+    For the Japan Pac region.
+    """
+
+    REQUIRES_SITES_PARAMETERS = {'vs30', 'z2pt5'}
+
+    def __init__(self):
+        super(ParkerEtAl2020SSlabJapanPac, self) \
+            .__init__(region="JP", saturation_region="JP_Pac")
+
+
+class ParkerEtAl2020SSlabJapanPhi(ParkerEtAl2020SSlab):
+    """
+    For the Japan Phi region.
+    """
+
+    REQUIRES_SITES_PARAMETERS = {'vs30', 'z2pt5'}
+
+    def __init__(self):
+        super(ParkerEtAl2020SSlabJapanPhi, self) \
+            .__init__(region="JP", saturation_region="JP_Phi")
+
+
+class ParkerEtAl2020SSlabSAN(ParkerEtAl2020SSlab):
+    """
+    For the SA North region.
+    """
+    def __init__(self):
+        super(ParkerEtAl2020SSlabSAN, self) \
+            .__init__(region="SA", saturation_region="SA_N")
+
+
+class ParkerEtAl2020SSlabSAS(ParkerEtAl2020SSlab):
+    """
+    For the SA South region.
+    """
+    def __init__(self):
+        super(ParkerEtAl2020SSlabSAS, self) \
+            .__init__(region="SA", saturation_region="SA_S")
+
+
+class ParkerEtAl2020SSlabTaiwanE(ParkerEtAl2020SSlab):
+    """
+    For the Taiwan East region.
+    """
+    def __init__(self):
+        super(ParkerEtAl2020SSlabTaiwanE, self) \
+            .__init__(region="TW", saturation_region="TW_E")
+
+
+class ParkerEtAl2020SSlabTaiwanW(ParkerEtAl2020SSlab):
+    """
+    For the Taiwan West region.
+    """
+    def __init__(self):
+        super(ParkerEtAl2020SSlabTaiwanW, self) \
+            .__init__(region="TW", saturation_region="TW_W")
```

Comment:
You do not need to define a lot of subclasses. You can use the aliases feature instead:
```python

from openquake.hazardlib.gsim.base import gsim_aliases

gsim_aliases['ParkerEtAl2020SInterAleutian'] = '''[ParkerEtAl2020SInter]
region="AK"
saturation_region="Aleutian"
'''
gsim_aliases['ParkerEtAl2020SInterAlaska'] = '''[ParkerEtAl2020SInter]
region="AK"
'''
gsim_aliases['ParkerEtAl2020SInterAleutian'] = '''[ParkerEtAl2020SInter]
region="AK"
saturation_region="Aleutian"
'''
gsim_aliases['ParkerEtAl2020SInterCAMN'] = '''[ParkerEtAl2020SInter]
region="CAM"
saturation_region="CAM_N"
'''
gsim_aliases['ParkerEtAl2020SInterCAMS'] = '''[ParkerEtAl2020SInter]
region="CAM"
saturation_region="CAM_S"
'''
...
```